prisma/schema.prisma

```diff
@@ -105,56 +119,59 @@ enum Status {
   CHECKED_IN
 }
 
+/// @zod.import(["import isMobilePhone from 'validator/lib/isMobilePhone'", "import { refineAgeRange, refineWordLength, refineIsTrue, transformStringToNullIfEmpty } from '../../../src/utils/zodValidators'"])
 model DH10Application {
-  id String @id @default(cuid())
+  // Make id optional so that frontend need not provide the id
+  id String @id @default(cuid()) /// @zod.custom.use(z.string().cuid().optional())
 
-  firstName String
-  lastName  String
+  firstName String /// @zod.string.min(1).max(255)
+  lastName  String /// @zod.string.min(1).max(255)
 
-  birthday DateTime
+  birthday DateTime /// @zod.custom.use(z.coerce.date().refine((date) => refineAgeRange(date, 13), { message: "You must be at least 13 years old" }))
 
   studyEnrolledPostSecondary Boolean
-  studyLocation              String?
-  studyDegree                String?
-  studyMajor                 String?
-  studyYearOfStudy           String?
+  studyLocation              String? /// @zod.custom.use(z.string().min(1).max(255).transform(transformStringToNullIfEmpty))
+  studyDegree                String? /// @zod.custom.use(z.string().min(1).max(255).transform(transformStringToNullIfEmpty))
+  studyMajor                 String? /// @zod.custom.use(z.string().min(1).max(255).transform(transformStringToNullIfEmpty))
+  studyYearOfStudy           String? /// @zod.custom.use(z.string().transform(transformStringToNullIfEmpty))
   studyExpectedGraduation    DateTime?
 
-  previousHackathonsCount Int
+  previousHackathonsCount Int /// @zod.custom.use(z.coerce.number().int().min(0))
 
-  longAnswerChange     String
-  longAnswerExperience String
-  longAnswerTech       String
-  longAnswerMagic      String
+  longAnswerChange     String /// @zod.custom.use(z.string().min(1, { message: "An answer is required for this question" }).refine((val) => refineWordLength(val, 150), { message: "Must be less than 150 words" }))
```

Comment:
I don't like this at all. We should keep the database abstraction and custom typing schema we need seperate.